src/notifiers/ntfy.py

```diff
@@ -0,0 +1,70 @@
+import json
+import logging
+
+import requests
+
+from models import Config, Item
+from models.errors import MaskConfigurationError, NtfyConfigurationError
+from notifiers import Notifier
+
+log = logging.getLogger('tgtg')
+
+
+class Ntfy(Notifier):
+    """Notifier for ntfy"""
+
+    def __init__(self, config: Config):
+        self.enabled = config.ntfy.get("enabled", False)
+        self.server = config.ntfy.get("server", "https://ntfy.sh")
+        self.topic = config.ntfy.get("topic")
+        self.title = config.ntfy.get("title", "tgtg")
+        self.body = config.ntfy.get("body")
+        self.priority = config.ntfy.get("priority", "default")
+        self.tags = config.ntfy.get("tags", "tgtg")
+        self.username = config.ntfy.get("username")
+        self.password = config.ntfy.get("password")
+        self.timeout = config.ntfy.get("timeout", 60)
+        self.cron = config.ntfy.get("cron")
+        if self.enabled and (not self.server or not self.topic):
+            raise NtfyConfigurationError()
+        if self.enabled:
+            try:
+                Item.check_mask(self.body)
+            except MaskConfigurationError as exc:
+                raise NtfyConfigurationError(exc.message) from exc
+
+    def send(self, item: Item) -> None:
+        """Sends item information via configured ntfy endpoint"""
+        if self.enabled and self.cron.is_now:
+            auth = None
+            if (self.username and self.password) is not None:
+                auth = requests.auth.HTTPBasicAuth(self.username, self.password)
+                log.info("Using basic auth with user '%s' for ntfy", self.username)
+            elif self.username is not None:
+                log.warning("Password missing for ntfy authentication, defaulting to no auth")
+            elif self.password is not None:
+                log.warning("Username missing for ntfy authentication, defaulting to no auth")
```

Comment:
Creating the `auth` object only once on initialization should be sufficient. Please move it to `__init__`.
Please only include necessary log messages to keep the console clean. You could also change it to `log.debug`.

Reply:
Good suggestion. :)